File: old-projects/vector-defence-silverlight-port/src/Monsters/TriangleMonster.ts
import { Point } from "../Point";
import { Monster } from "./Monster";

export class TriangleMonster extends Monster {
  constructor(x: number, y: number, path: Point[]) {
    super(x, y, path, 1.75);
    this.bounty = 30;
    this.color = "#ffa500";
    this.radius = 7;
    this.life = this.originalLife = 100;
  }

  override render(context: CanvasRenderingContext2D): void {
    context.save();
    context.translate(this.x, this.y);
    context.rotate(this.angle);
    context.fillStyle = this.damageAlpha > 0 ? `rgba(128,0,128,${this.damageAlpha})` : "#000";
    context.strokeStyle = this.color;
    context.lineWidth = 1.5;
    context.beginPath();
    context.moveTo(-6, -6);
    context.lineTo(6, 0);
    context.lineTo(-6, 6);
    context.closePath();
    context.fill();
    context.stroke();
    context.restore();
    super.render(context);
  }
}

File: silverlight-ports/vector-defence/original-csharp-source/Monsters/TriangleMonster.cs
using System.Collections.Generic;
using System.Windows;
using System.Windows.Media;
using System.Windows.Shapes;

namespace VectorDefenceSL.Monsters
{
    public class TriangleMonster : Monster
    {
        private readonly RotateTransform transform = new RotateTransform();

        public TriangleMonster(double x, double y, IList<Point> path) : base(x, y, path, 1.75)
        {
            Bounty = 30;
            Color = Colors.Orange;
            Radius = 7;
            life = originalLife = 100;
            mainElement = new Polygon
            {
                StrokeThickness = 1.5,
                Fill = new SolidColorBrush(Colors.Black),
                Stroke = new SolidColorBrush(Color),
                RenderTransform = transform,
                Points = new PointCollection
                {
                    new Point(-6, -6),
                    new Point(6, 0),
                    new Point(-6, 6),
                },
            };
            Game.AddElement(mainElement);
        }

        public override void Render()
        {
            Move(X, Y);
            base.Render();
        }

        protected override void AngleChanged(double newAngle)
        {
            transform.Angle = Util.GetAngleInDegrees(newAngle);
        }
    }
}